### PR TITLE
Add the ability to generate Tls certificate with QED command tool

### PR DESCRIPTION
**Pull request by suizman**
Improve our generate command in **QED** command tool and enable it to generate self-signed **Tls** certificate.
This feature allow us to remove **openssl** (external|sys tools) dependency in our test scripts.

### CODE REVIEW

File: cmd/generate_certs.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+func init() {
+	generateCmd.AddCommand(generateTlsCerts)
+}
+
+func runGenerateTlsCerts(cmd *cobra.Command, args []string) {
+
+	conf := generateCtx.Value(k("generate.config")).(*GenerateConfig)
+
+	cert, key, err := crypto.NewTlsCerts(conf.Path, conf.Host)
```
Comment (jllucas) [inline cmd/generate_certs.go]: We should add URL checking to conf.Host, as done in other cmd commands (agent, server, client,...).

File: crypto/certs.go [hunk 2]
```diff
@@ -28,10 +28,10 @@ import (
 	"time"
 )
 
-func GenerateTlsCert(path string) (string, error) {
+func NewTlsCerts(path, host string) (string, string, error) {
```
Comment (jllucas) [inline crypto/certs.go]: Add godoc to crypto package and methods.
It is difficult to guess the function output (string, string,...).

File: tests/build_certs [hunk 3]
```diff
@@ -10,52 +24,31 @@ function _readlink() { (
...
+
 sign_path=${pub}
-cert_path=${pub}/server.crt
-key_path=${pub}/server.key
+cert_path=${pub}
 
 (
 cd ${tdir}
 
+if [ ! -f ${node_path} ]; then (
```
Comment (jllucas) [inline tests/build_certs]: I don't understand why node-exporter is here, in "build_certs" script.

**Comment by gdiazlo** [comment]
I would love the commands be like:

`qed generate self-signed-cert` -> this is the command as it is now
`qed generate csr` -> generates a  csr to be signed by an external CA

File: tests/build_certs (hunk 3 above)
Comment (suizman) [reply tests/build_certs]: Fixed! It was a copy & paste typo

**Comment by suizman** [comment]
> I would love the commands be like:
> 
> `qed generate self-signed-cert` -> this is the command as it is now
> `qed generate csr` -> generates a csr to be signed by an external CA

Renaming done and you can also check my proposal for CSR generation  in https://github.com/BBVA/qed/pull/138/commits/206e8711900c0b5b5b5e4dbac2763a53afbd50d7

**Pull request merged**